test/neighborsearch.jl

```diff
@@ -0,0 +1,59 @@
+@testset "Neighborsearch" begin
+  @testset "KBallSearcher" begin
+    # KBallSearcher
+
+    #create a regular grid to test
+    pdomain = RegularGrid(10,10)
+
+    #test 1: all points should be found
+    k = 10
+    ball = BallNeighborhood(100.0)
+    kballsearcher = KBallSearcher(pdomain, k, ball)
+
+    neighbors = neighbors = Vector{Int}(undef, k)
+    xₒ = [5.0,5.0]
+    nneigh = search!(neighbors,xₒ,kballsearcher; mask=nothing)
+
+    @test nneigh == 10
+
+    #test 2: radius of 1.0 m 5 points should be found
+    k = 10
+    ball = BallNeighborhood(1.0)
+    kballsearcher = KBallSearcher(pdomain, k, ball)
+
+    neighbors = Vector{Int}(undef, k)
+    xₒ = [5.0,5.0]
+    nneigh = search!(neighbors,xₒ,kballsearcher; mask=nothing)
```

Comment:
sample applies to here, you could do `neighbors = search(xo, kballsearcher)` and then `nneigh = length(neighbors)`